patches/api/0457-Paper-Item-API.patch

```diff
@@ -0,0 +1,582 @@
++
++    // AxolotlBucket
++    PropertyKey<Axolotl.Variant> VARIANT = PropertyKey.of("Variant", Axolotl.Variant.class);
++
++    // MusicInstrumentMeta
++    PropertyKey<MusicInstrument> INSTRUMENT = PropertyKey.of("instrument", MusicInstrument.class);
++
++    // BannerMeta
++    PropertyKey<DyeColor> SHIELD_BANNER_BASE = PropertyKey.of("Base", DyeColor.class);
++
++}
+diff --git a/src/main/java/io/papermc/paper/inventory/item/properties/StewEffect.java b/src/main/java/io/papermc/paper/inventory/item/properties/StewEffect.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..a3232752c57b0b9a6c1ae7aac429166a2db987b2
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/inventory/item/properties/StewEffect.java
+@@ -0,0 +1,25 @@
++package io.papermc.paper.inventory.item.properties;
++
++import org.bukkit.potion.PotionEffectType;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Represents a serialized stew effect with a duration.
++ */
++public interface StewEffect {
++
++    /**
++     * Gets the effect type
++     * @return type
++     */
++    @NotNull PotionEffectType effect();
++
++    /**
++     * Gets the duration for this stew effect instnace.
++     * By default, this will be 160 ticks.
++     *
++     * @return duration override
++     */
++    int duration();
++}
+diff --git a/src/main/java/io/papermc/paper/inventory/item/properties/TropicalFishVariant.java b/src/main/java/io/papermc/paper/inventory/item/properties/TropicalFishVariant.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..3ae8e35cb98c2b39714f3347ad49a474e505b45a
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/inventory/item/properties/TropicalFishVariant.java
+@@ -0,0 +1,29 @@
++package io.papermc.paper.inventory.item.properties;
++
++import org.bukkit.DyeColor;
++import org.bukkit.entity.TropicalFish;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Represents a serialized tropical fish.
++ */
++public interface TropicalFishVariant {
++
++    /**
++     * Gets the pattern color
++     * @return color
++     */
++    @NotNull DyeColor patternColor();
++
++    /**
++     * Gets the body color
++     * @return color
++     */
++    @NotNull DyeColor bodyColor();
++
++    /**
++     * Gets the type
++     * @return type
++     */
++    @NotNull TropicalFish.Pattern type();
++}
+diff --git a/src/main/java/io/papermc/paper/property/MutablePropertyView.java b/src/main/java/io/papermc/paper/property/MutablePropertyView.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..d73a2276a3e56905e9595f85f5532f06f8dc583d
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/property/MutablePropertyView.java
+@@ -0,0 +1,56 @@
++package io.papermc.paper.property;
++
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++public interface MutablePropertyView extends PropertyView, Cloneable {
```

Comment:
This concept should not be introduced in the API. It doesn't make sense to have MutablePropertyView properties which can set any random key on them.